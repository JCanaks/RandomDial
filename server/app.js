import express from 'express';
import cors from 'cors';
import morgan from 'morgan';
import 'dotenv/config';
import bodyParser from 'body-parser';
import routes from './routes';

const app = express();
app.use(cors());
app.use(bodyParser.urlencoded({ extended: false }));
app.use(bodyParser.json());
app.use(morgan('dev'));
app.use('/api', routes);

app.get('/', (req, res) => {
  res.send({
    message: 'Welcome to RandomDail, your Simple Random Phone Number Generator',
  });
});

app.use((err, req, res, next) => res.status.json({
  message: 'Internal Server error',
}));

export default app;
