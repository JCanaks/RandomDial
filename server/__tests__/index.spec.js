import '@babel/polyfill';
import request from 'supertest';
import app from '../app';

describe('Testing the default Home API', () => {
    test('Default Home API with welcome message', async () => {
        const response = await request(app).get('/');
        expect(response.body).toMatchSnapshot();;
        expect(response.statusCode).toBe(200);
    })
});
