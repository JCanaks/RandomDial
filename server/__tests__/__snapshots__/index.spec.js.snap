// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the default Home API Default Home API with welcome message 1`] = `
Object {
  "message": "Welcome to RandomDail, your Simple Random Phone Number Generator",
}
`;
